### PR TITLE
releases: update TiDB v3.0.17 release notes

### PR DESCRIPTION
**Pull request by ti-srebot**
update TiDB v3.0.17 release notes

### CODE REVIEW

File: releases/3.0.17.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    - Ease the impact of stats feedback on cluster [#18770](https://github.com/pingcap/tidb/pull/18770)
+    - Limit batch split count for one request [#18694](https://github.com/pingcap/tidb/pull/18694)
+    - Fix the issue of request `/tiflash/replica` HTTP API was slow when there are many history DDL jobs in TiDB cluster. [#18386](https://github.com/pingcap/tidb/pull/18386)
+    - Improve row count estimation for index equal condition [#17609](https://github.com/pingcap/tidb/pull/17609)
+
++ TiKV
+
+    - Add `hibernate-timeout` config that allows region being hibernated later to improve rolling update performance [#8207](https://github.com/tikv/tikv/pull/8207)
+
```
Comment (overvenus) [inline releases/3.0.17.md:38]:
```suggestion

+ TiDB-Lightning

    - `[black-white-list]` has been deprecated with a newer, easier-to-understand filter format. [#332](https://github.com/pingcap/tidb-lightning/pull/332)

```

File: releases/3.0.17.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    - Return the actual error message when a query connection panics. [#18498](https://github.com/pingcap/tidb/pull/18498)
+    - Fix unknown column error for sql like `select a from t having t.a` [#18432](https://github.com/pingcap/tidb/pull/18432)
+    - Fix add a primary key when the table's  pk_is_handle is true [#18342](https://github.com/pingcap/tidb/pull/18342)
+    - Check plan type when executing `explain format="dot"`. [#17157](https://github.com/pingcap/tidb/pull/17157)
+
++ TiKV
+
+    - Fix a bug that may read stale data during region merging [#8111](https://github.com/tikv/tikv/pull/8111)
+
```
Comment (overvenus) [inline releases/3.0.17.md:25]:
```suggestion

+ TiDB-Lightning

    - Fix the issue that the `log-file` flag is ignored. [#345](https://github.com/pingcap/tidb-lightning/pull/345)

```

**Comment by XuHuaiyu** [comment]
Modification for TiDB:
```
## Bug Fixes

+ TiDB

    - Return the actual error message instead of empty set when a query which contains IndexHashJoin or IndexMergeJoin encounter a panic [#18498](https://github.com/pingcap/tidb/pull/18498)
    - Fix unknown column error for sql like `select a from t having t.a` [#18432](https://github.com/pingcap/tidb/pull/18432)
    - Fordid adding primary key for a table when the table has no primary key or the table has an integer primary key already. [#18342](https://github.com/pingcap/tidb/pull/18342)
    - Return empty set when executing `explain format="dot" for connection` [#17157](https://github.com/pingcap/tidb/pull/17157)
    - Fix STR_TO_DATE's handling for format token '%r', '%h' [#18725](https://github.com/pingcap/tidb/pull/18725)

## Improvements

+ TiDB

    - Ease the impact of stats feedback on cluster [#18770](https://github.com/pingcap/tidb/pull/18770)
    - Limit batch split count for one request [#18694](https://github.com/pingcap/tidb/pull/18694)
    - Acceleate `/tiflash/replica` HTTP API when there are many history DDL jobs in TiDB cluster [#18386](https://github.com/pingcap/tidb/pull/18386)
    - Improve row count estimation for index equal condition [#17609](https://github.com/pingcap/tidb/pull/17609)
    - Speed up the executing of `kill tidb conn_id` [#18506](https://github.com/pingcap/tidb/pull/18506)
```

File: releases/release-3.0.17.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
++ TiKV
+
+    - Add `hibernate-timeout` config that allows region being hibernated later to improve rolling update performance [#8207](https://github.com/tikv/tikv/pull/8207)
+
++ TiDB-Lightning
+
+    - `[black-white-list]` has been deprecated with a newer, easier-to-understand filter format. [#332](https://github.com/pingcap/tidb-lightning/pull/332)
+
+## Others
+
```
Comment (ran-huang) [inline releases/release-3.0.17.md:48]: @uglyengineer Please help confirm which categories these notes belong to. Thanks~

Comment (uglyengineer) [reply releases/release-3.0.17.md:48]: All of the three can be deleted (1&2 already moved up, 3 should not be here.)

**Comment by XuHuaiyu** [comment]
LGTM

File: releases/release-3.0.17.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    - Fix the unknown column error for SQL statements like `SELECT a FROM t HAVING t.a` [#18432](https://github.com/pingcap/tidb/pull/18432)
+    - Forbid adding a primary key for a table when the table has no primary key or when the table already has an integer primary key [#18342](https://github.com/pingcap/tidb/pull/18342)
+    - Return an empty set when executing `EXPLAIN FORMAT="dot" FOR CONNECTION` [#17157](https://github.com/pingcap/tidb/pull/17157)
+    - Fix `STR_TO_DATE`'s handling for format token '%r', '%h' [#18725](https://github.com/pingcap/tidb/pull/18725)
+
++ TiKV
+
+    - Fix a bug that might read stale data during region merging [#8111](https://github.com/tikv/tikv/pull/8111)
+
++ TiDB-Lightning
```
Comment (ran-huang) [inline releases/release-3.0.17.md:25]:
```suggestion
+ TiDB Lightning
```

File: releases/release-3.0.17.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    - Limit batch split count for one request [#18694](https://github.com/pingcap/tidb/pull/18694)
+    - Accelerate `/tiflash/replica` HTTP API when there are many history DDL jobs in the TiDB cluster [#18386](https://github.com/pingcap/tidb/pull/18386)
+    - Improve row count estimation for index equal condition [#17609](https://github.com/pingcap/tidb/pull/17609)
+    - Speed up the execution of `kill tidb conn_id` [#18506](https://github.com/pingcap/tidb/pull/18506)
+
++ TiKV
+
+    - Add `hibernate-timeout` config that allows region being hibernated later to improve rolling update performance [#8207](https://github.com/tikv/tikv/pull/8207)
+
++ TiDB-Lightning
```
Comment (ran-huang) [inline releases/release-3.0.17.md:43]:
```suggestion
+ TiDB Lightning
```

File: releases/release-3.0.17.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+    - Ease the impact of statistics feedback on the cluster [#18770](https://github.com/pingcap/tidb/pull/18770)
+    - Limit batch split count for one request [#18694](https://github.com/pingcap/tidb/pull/18694)
+    - Accelerate `/tiflash/replica` HTTP API when there are many history DDL jobs in the TiDB cluster [#18386](https://github.com/pingcap/tidb/pull/18386)
+    - Improve row count estimation for index equal condition [#17609](https://github.com/pingcap/tidb/pull/17609)
+    - Speed up the execution of `kill tidb conn_id` [#18506](https://github.com/pingcap/tidb/pull/18506)
+
++ TiKV
+
+    - Add `hibernate-timeout` config that allows region being hibernated later to improve rolling update performance [#8207](https://github.com/tikv/tikv/pull/8207)
```
Comment (ran-huang) [inline releases/release-3.0.17.md:41]:
```suggestion
    - Add the `hibernate-timeout` configuration that delays region hibernation to improve rolling update performance [#8207](https://github.com/tikv/tikv/pull/8207)
```

Comment (ran-huang) [reply releases/release-3.0.17.md:41]: @XuHuaiyu PTAL

**Comment by XuHuaiyu** [comment]
LGTM

File: releases/release-3.0.17.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
+---
+title: TiDB 3.0.17 Release Notes
+---
+
+# TiDB 3.0.17 Release Notes
+
+Release date: Aug 03, 2020
```
Comment (TomShawn) [inline releases/release-3.0.17.md:7]:
```suggestion
Release date: Aug 3, 2020
```

File: releases/release-3.0.17.md [hunk 8]
```diff
@@ -21,6 +21,7 @@ TiDB version: 3.0.17
 + TiKV
 
     - Fix a bug that might read stale data during region merging [#8111](https://github.com/tikv/tikv/pull/8111)
+    - Fix memory leak during scheduling [#8355](https://github.com/tikv/tikv/pull/8355)
```
Comment (TomShawn) [inline releases/release-3.0.17.md:24]:
```suggestion
    - Fix the issue of memory leak during the scheduling process [#8355](https://github.com/tikv/tikv/pull/8355)
```

**Pull request merged**